fmp_server.py

```diff
@@ -39,9 +39,833 @@ def recvAnsConnection(self, packet_id, data, seq):
         self.server.debug("Connection: {!r}".format(connection_data))
         self.sendNtcLogin(3, connection_data, seq)
 
+    def sendAnsLayerDown(self, layer_id, layer_set, seq):
```

Comment:
I'm not a huge fan of shadowing existing methods like that, even if it makes merge-conflict less likely it also tends to let bugs go through more easily. You can keep it as is, atm. I'll probably try to think of a cleaner approach. 

Basically, my original thought was to let the server logic and specialised server logic in servers modules. Then, the serialisation and PAT logic to their own modules.